### ISSUE TITLE
0.6.1 does not finish properly

### ISSUE THREAD

**Issue by beto-rodriguez** [Redth/dotnet-maui-check#69]
I seems that when the installation gets paused by an error in version 0.6.1 then `maui-check` never completes.

I was installing using v 0.6.1, In got a warning about not having an emulator installed, I installed and now it never fished the installation again when I ran `maui-check`.

I rolled back to 0.6.0 and it completed properly.

To reproduce, just run `maui-check` once it is already installed and you will see the error:

> ! There were one or more problems detected.
Please review the errors and correct them and run maui-check again.
(NOTE: there were not listed errors)

Now roll back to `0.6.0` and it will let you know that your machine is ready:

> Congratulations, everything looks great!

**Comment by JunkyXL86** [comment]
Same here. Every check is green, but in the end you'll see an error.

```
      _   _   _____   _____     __  __      _      _   _   ___
     | \ | | | ____| |_   _|   |  \/  |    / \    | | | | |_ _|
     |  \| | |  _|     | |     | |\/| |   / _ \   | | | |  | |
  _  | |\  | | |___    | |     | |  | |  / ___ \  | |_| |  | |
 (_) |_| \_| |_____|   |_|     |_|  |_| /_/   \_\  \___/  |___|

¤ .NET MAUI Check v0.6.1.0 ¤
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
This tool will attempt to evaluate your .NET MAUI development environment.
If problems are detected, this tool may offer the option to try and fix them for you, or suggest a way to fix them
yourself.

Thanks for choosing .NET MAUI!
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
» Synchronizing configuration... ok
» Scheduling appointments... ok

> OpenJDK 11.0 Checkup...
  - 11.0.10 (C:\Program Files\Microsoft\jdk-11.0.10.9-hotspot\bin\..)
  - 1.8.0-25 (C:\Program Files\Android\Jdk\microsoft_dist_openjdk_1.8.0.25)

> Visual Studio 17.0.0-pre.2.0 Checkup...
  - 16.10.0
  - 16.11.0-pre.1.0

> Android SDK Checkup...
  - emulator (30.6.5)
  - build-tools;30.0.2 (30.0.2)
  - platforms;android-30 (3)
  - system-images;android-30;google_apis;x86 (9)
  - platform-tools (31.0.0)

> Android Emulator Checkup...
  - Emulator: nexus_5x_pie_9_0_-_api_28 found.

> .NET SDK Checkup...
  - 2.1.200 - C:\Program Files\dotnet\sdk\2.1.200
  - 2.1.524 - C:\Program Files\dotnet\sdk\2.1.524
  - 2.1.604 - C:\Program Files\dotnet\sdk\2.1.604
  - 2.1.700 - C:\Program Files\dotnet\sdk\2.1.700
  - 2.1.701 - C:\Program Files\dotnet\sdk\2.1.701
  - 2.2.107 - C:\Program Files\dotnet\sdk\2.2.107
  - 2.2.204 - C:\Program Files\dotnet\sdk\2.2.204
  - 2.2.300 - C:\Program Files\dotnet\sdk\2.2.300
  - 2.2.301 - C:\Program Files\dotnet\sdk\2.2.301
  - 3.1.410 - C:\Program Files\dotnet\sdk\3.1.410
  - 5.0.103 - C:\Program Files\dotnet\sdk\5.0.103
  - 5.0.300-preview.21258.4 - C:\Program Files\dotnet\sdk\5.0.300-preview.21258.4
  - 5.0.300 - C:\Program Files\dotnet\sdk\5.0.300
  - 5.0.301 - C:\Program Files\dotnet\sdk\5.0.301
  - 6.0.100-preview.6.21355.2 - C:\Program Files\dotnet\sdk\6.0.100-preview.6.21355.2

> .NET SDK - Workload Deduplication Checkup...

> Edge WebView2 Checkup...
  - Found Edge WebView2 version 91.0.864.67

> .NET SDK - Workloads (6.0.100-preview.6.21355.2) Checkup...
  - microsoft-android-sdk-full (Microsoft.NET.Sdk.Android.Manifest-6.0.100 : 30.0.100-preview.6.62) installed.
  - microsoft-ios-sdk-full (Microsoft.NET.Sdk.iOS.Manifest-6.0.100 : 15.0.100-preview.6.63) installed.
  - microsoft-maccatalyst-sdk-full (Microsoft.NET.Sdk.MacCatalyst.Manifest-6.0.100 : 15.0.100-preview.6.63) installed.
  - microsoft-tvos-sdk-full (Microsoft.NET.Sdk.tvOS.Manifest-6.0.100 : 15.0.100-preview.6.63) installed.
  - microsoft-macos-sdk-full (Microsoft.NET.Sdk.macOS.Manifest-6.0.100 : 12.0.100-preview.6.63) installed.
  - maui (Microsoft.NET.Sdk.Maui.Manifest-6.0.100 : 6.0.100-preview.6.1003) installed.
  - microsoft-net-runtime-android (microsoft.net.workload.mono.toolchain.manifest-6.0.100 : 6.0.0-preview.6.21352.12)
installed.
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────


**! There were one or more problems detected.
Please review the errors and correct them and run maui-check again.**

Press enter to exit...
```

**Comment by JunkyXL86** [comment]
So I updated to 2019 Preview 3 AND installed VS 2022 Preview .. that did it. I think Preview 3 was the more obvious cause.

**Comment by thluiz** [comment]
Installing only VS 2022 Preview fixed it here

**Comment by hofi199410** [comment]
> So I updated to 2019 Preview 3 AND installed VS 2022 Preview .. that did it. I think Preview 3 was the more obvious cause.

I have installed both versions, unfortunately, it still does not work. Do you have any other ideas?

**Comment by thluiz** [comment]
> > So I updated to 2019 Preview 3 AND installed VS 2022 Preview .. that did it. I think Preview 3 was the more obvious cause.
> 
> I have installed both versions, unfortunately, it still does not work. Do you have any other ideas?

I've needed to uninstall VS 2019 and VS 2019 Preview. It looks like the ordering of the installation makes a difference: If you install VS 2022 before VS2019 it'll work.

After maui-check passed I've needed to install this extension: https://marketplace.visualstudio.com/items?itemName=ProjectReunion.MicrosoftSingleProjectMSIXPackagingToolsDev17 (It only installed from the link. In VS2022's Extensions menu it doesn't work)
